parser/src/command/decision.rs

```diff
@@ -0,0 +1,214 @@
+
+/// A command as parsed and received from calling the bot with some arguments,
+/// like `@rustbot merge`
+#[derive(Debug, Eq, PartialEq)]
+pub struct DecisionCommand {
+    pub resolution: Resolution,
+    pub reversibility: Reversibility,
+    pub team: Option<String>,
+}
+
+impl DecisionCommand {
+    pub fn parse<'a>(input: &mut Tokenizer<'a>) -> Result<Option<Self>, Error<'a>> {
+        let mut toks = input.clone();
+
+        match toks.peek_token()? {
+            Some(Token::Word("merge")) => {
+                toks.next_token()?;
+
+                let team: Option<String> = get_team(&mut toks)?;
+
+                command_or_error(
+                    input,
+                    &mut toks,
+                    Self {
+                        resolution: Resolution::Merge,
+                        reversibility: Reversibility::Reversible,
+                        team,
+                    },
+                )
+            }
+            Some(Token::Word("hold")) => {
+                toks.next_token()?;
+
+                let team: Option<String> = get_team(&mut toks)?;
+
+                command_or_error(
+                    input,
+                    &mut toks,
+                    Self {
+                        resolution: Resolution::Hold,
+                        reversibility: Reversibility::Reversible,
+                        team,
+                    },
+                )
+            }
+            _ => Ok(None),
+        }
+    }
+}
+
+fn get_team<'a>(toks: &mut Tokenizer<'a>) -> Result<Option<String>, Error<'a>> {
+    match toks.peek_token()? {
+        Some(Token::Word(team)) => {
+            toks.next_token()?;
+
+            Ok(Some(team.to_string()))
+        }
+        _ => Ok(None),
+    }
+}
+
+fn command_or_error<'a>(
+    input: &mut Tokenizer<'a>,
+    toks: &mut Tokenizer<'a>,
+    command: DecisionCommand,
+) -> Result<Option<DecisionCommand>, Error<'a>> {
+    if let Some(Token::Dot) | Some(Token::EndOfLine) = toks.peek_token()? {
+        *input = toks.clone();
+        Ok(Some(command))
+    } else {
+        Err(toks.error(ParseError::ExpectedEnd))
+    }
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum ParseError {
+    ExpectedEnd,
+}
+
+impl std::error::Error for ParseError {}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ParseError::ExpectedEnd => write!(f, "expected end of command"),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, ToSql, FromSql, Eq, PartialEq)]
+#[postgres(name = "reversibility")]
+pub enum Reversibility {
+    #[postgres(name = "reversible")]
+    Reversible,
+    #[postgres(name = "irreversible")]
+    Irreversible,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, ToSql, FromSql, Eq, PartialEq)]
+#[postgres(name = "resolution")]
+pub enum Resolution {
+    #[postgres(name = "merge")]
+    Merge,
+    #[postgres(name = "hold")]
+    Hold,
```

Comment:
Agree on adding close but we should keep hold as the documentation we based our impl specifies https://lang-team.rust-lang.org/decision_process/examples.html, because is the way to "pause" the process. 

In the current code it's not doing that so we will fix it as well

Reply:
I'm fine with extending the process a little per the proposal, though let's avoid delaying too much as a result.